optuna/integration/dask.py

```diff
@@ -0,0 +1,738 @@
+    ) -> None:
+        return self.get_storage(storage_name).set_trial_value(
+            trial_id=trial_id,
+            value=value,
+        )
+
+    def set_trial_intermediate_value(
+        self,
+        comm: "distributed.comm.tcp.TCP",
+        storage_name: str,
+        trial_id: int,
+        step: int,
+        intermediate_value: float,
+    ) -> None:
+        return self.get_storage(storage_name).set_trial_intermediate_value(
+            trial_id=trial_id,
+            step=step,
+            intermediate_value=intermediate_value,
+        )
+
+    def set_trial_user_attr(
+        self,
+        comm: "distributed.comm.tcp.TCP",
+        storage_name: str,
+        trial_id: int,
+        key: str,
+        value: Any,
+    ) -> None:
+        return self.get_storage(storage_name).set_trial_user_attr(
+            trial_id=trial_id,
+            key=key,
+            value=value,
+        )
+
+    def set_trial_system_attr(
+        self,
+        comm: "distributed.comm.tcp.TCP",
+        storage_name: str,
+        trial_id: int,
+        key: str,
+        value: Any,
+    ) -> None:
+        return self.get_storage(storage_name).set_trial_system_attr(
+            trial_id=trial_id,
+            key=key,
+            value=value,
+        )
+
+    def get_trial(
+        self,
+        comm: "distributed.comm.tcp.TCP",
+        storage_name: str,
+        trial_id: int,
+    ) -> dict:
+        trial = self.get_storage(storage_name).get_trial(trial_id=trial_id)
+        return _serialize_frozentrial(trial)
+
+    def get_all_trials(
+        self,
+        comm: "distributed.comm.tcp.TCP",
+        storage_name: str,
+        study_id: int,
+        deepcopy: bool = True,
+    ) -> List[dict]:
+        trials = self.get_storage(storage_name).get_all_trials(
+            study_id=study_id,
+            deepcopy=deepcopy,
+        )
+        return [_serialize_frozentrial(t) for t in trials]
+
+    def get_n_trials(
+        self,
+        comm: "distributed.comm.tcp.TCP",
+        storage_name: str,
+        study_id: int,
+        state: Optional[TrialState] = None,
+    ) -> int:
+        return self.get_storage(storage_name).get_n_trials(
+            study_id=study_id,
+            state=state,
+        )
+
+    def read_trials_from_remote_storage(
+        self,
+        comm: "distributed.comm.tcp.TCP",
+        storage_name: str,
+        study_id: int,
+    ) -> None:
+        return self.get_storage(storage_name).read_trials_from_remote_storage(study_id=study_id)
+
+
+def _register_with_scheduler(
+    dask_scheduler: "distributed.Scheduler", storage: str, name: str
+) -> None:
+    if "optuna" not in dask_scheduler.extensions:
+        ext = _OptunaSchedulerExtension(dask_scheduler)
+    else:
+        ext = dask_scheduler.extensions["optuna"]
+
+    if name not in ext.storages:
+        ext.storages[name] = optuna.storages.get_storage(storage)
+
+
+def _use_basestorage_doc(func: Callable) -> Callable:
```

Comment:
It's certainly not required. I primarily added this to make adding docstrings to `DaskStorage` easy. I'm happy to remove it, or move it to a more generic utilities module if you think this could be useful in other places.